controllers/openstackdataplanenodeset_controller.go

```diff
@@ -388,6 +389,21 @@ func (r *OpenStackDataPlaneNodeSetReconciler) SetupWithManager(mgr ctrl.Manager)
 			}
 			result = append(result, reconcile.Request{NamespacedName: name})
 		}
+		podsInterface := r.Kclient.CoreV1().Pods(namespace)
+		// List the pods in the given namespace
+		podsList, err := podsInterface.List(context.TODO(), v1.ListOptions{})
```

Comment:
you can filter by deployment: https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/pkg/util/ansible_execution.go#L68

and you probably need to add it:
```
//+kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch
```
https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/controllers/openstackdataplanenodeset_controller.go#L120

Reply:
you can also use the `field-selector` https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/ to filter `status.phase=Failed`

Reply:
Thanks, that does make sense.